bluemira/equilibria/coils/_grouping.py

```diff
@@ -998,3 +1001,26 @@ def _sum(
         ind = self._control_ind if control else slice(None)
 
         return np.sum(output[..., ind], axis=-1) if sum_coils else output[..., ind]
+
+    def get_coiltype(self, ctype):
+        """Get coils by coils type"""
+        if coiltype := self._get_coiltype(ctype):
+            return CoilSet(*coiltype)
+        return None
+
+    @classmethod
+    def from_group_vecs(cls, eqdsk: EQDSKInterface):
+        """Same as its parent class,
+        initialize an instance of CoilSet from group vectors,
+        but it also automatically decide which coils are control coils
+        according to CoilType.get_actively_controlled_types.
+        """
+        self = super().from_group_vecs(eqdsk)
+
+        controlled_coils = []
+        for active_coiltype in CoilType.get_actively_controlled_types():
+            controlled_coils.extend(
+                [coil.name for coil in self._get_coiltype(active_coiltype)]
+            )
+        self.control = controlled_coils
```

Comment:
Its only setting the one variable really so I prefer inlining the assignment (list comprehension reverses the order of the loops)
```suggestion
        self.control = [
            coil.name
            for active_coiltype in (CoilType.CS, CoilType.PF)
            for coil in self._get_coiltype(active_coiltype)
        ]
```